daemon/container.go

```diff
@@ -1557,3 +1548,37 @@ func (c *Container) LogDriverType() string {
 	}
 	return c.hostConfig.LogConfig.Type
 }
+
+type archiveFilter struct {
+	basePath string
+	files    []string
+}
+
+func archiveGlob(basePath string) (*archiveFilter, error) {
+	all, err := filepath.Glob(basePath)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(all) == 0 {
+		return nil, os.ErrNotExist
+	}
+
+	baseDir := filepath.Dir(basePath)
+	for baseDir != "" && !strings.HasPrefix(all[0], baseDir) {
```

Comment:
yeah, I think you're right. The problem here is that there are not enough tests to cover other behaviors with more than one file. I'll change the logic to reflect that.